packages/x-date-pickers/src/AdapterLuxon/AdapterLuxon.ts

```diff
@@ -356,13 +371,13 @@ export class AdapterLuxon implements MuiPickersAdapter<DateTime, string> {
 
   public isBeforeYear = (value: DateTime, comparing: DateTime) => {
     const comparingInValueTimezone = this.setTimezone(comparing, this.getTimezone(value));
-    const diff = value.diff(comparingInValueTimezone.startOf('year'), 'years').toObject();
+    const diff = value.diff(this.startOfYear(comparingInValueTimezone), 'years').toObject();
     return diff.years! < 0;
   };
 
   public isBeforeDay = (value: DateTime, comparing: DateTime) => {
     const comparingInValueTimezone = this.setTimezone(comparing, this.getTimezone(value));
-    const diff = value.diff(comparingInValueTimezone.startOf('day'), 'days').toObject();
+    const diff = value.diff(this.startOfDay(comparingInValueTimezone), 'days').toObject();
```

Comment:
Not directly related but I moved all the `startOf` / `endOf` to use our custom methods in case we have other additional behaviors in the future